stan/math/rev/fun/matrix_power.hpp

```diff
@@ -59,7 +59,7 @@ class matrix_product_vari_n1 : public vari {
 };
 
 template <int R, int C>
-class matrix_product_vari : public vari {
+class matrix_product_vari : public vari, public chainable_alloc {
```

Comment:
This allocates a `std::vector` of things in it, so it needs to be on the chainable stack to get cleaned up.